### PR TITLE
Disable buy item by default

### LINKED ISSUE #1220: Disable !buyitem by default
PLEASE just do it, one of my friends just lost 15k robux worth of UGC due to this.

### PR DESCRIPTION
**Pull request by RadiantUwU**
Fixes #1220

### CODE REVIEW

**Comment by LiveFireExercise** [comment]
You need to add these changes to defaultsettings as well

**Comment by RadiantUwU** [comment]
> You need to add these changes to defaultsettings as well

Oh? Hold on.

**Comment by RadiantUwU** [comment]
Donezies

**Comment by WalkerOfBacon** [comment]
will test

**Review by WalkerOfBacon — APPROVED** [review]

**Comment by LiveFireExercise** [comment]
Why do we need this tho

**Comment by RadiantUwU** [comment]
> Why do we need this tho

#1220 

Many UGC games are vulnerable due to not knowing about this.

**Review by moo1210 — APPROVED** [review]
It was initially very unclear what this vulnerability was due to pretty undetailed information, but I now understand this is due to the feature of awarding limited UGCs in-game. I don't know why this command even exists in the first place.

**Review by fxeP1 — APPROVED** [review]
LGTM

**Comment by P3tray** [comment]
> It was initially very unclear what this vulnerability was due to pretty undetailed information, but I now understand this is due to the feature of awarding limited UGCs in-game. I don't know why this command even exists in the first place.

Could you give a more clear explanation of the problem too?

**Comment by LiveFireExercise** [comment]
Why is it by default tho

**Comment by RadiantUwU** [comment]
> Why is it by default tho

Good question, !buyitem was created before limited UGCs appeared

**Comment by LiveFireExercise** [comment]
why is it turned off by default instead of on by default

**Comment by RadiantUwU** [comment]
> why is it turned off by default instead of on by default

This pull request has been made to disable it by default due to its continuous exploitation in games with limited UGCs creating a security vulnerability inside the module.

**Comment by RadiantUwU** [comment]
> why is it turned off by default instead of on by default

This pull request has been made to disable it by default due to its continuous exploitation in games with limited UGCs creating a security vulnerability inside the module.

**Comment by LiveFireExercise** [comment]
Okay as part of the Epix Testing Initiative, I am going to test this before merging it

File: MainModule/Server/Commands/Players.lua [hunk 1]
```diff
@@ -1150,6 +1150,7 @@ return function(Vargs, env)
 			Args = {"id"};
 			Description = "Prompts yourself to buy the asset belonging to the ID supplied";
 			AdminLevel = "Players";
+			Disabled = Settings.DisableBuyItem ~= false;
```
Comment (LiveFireExercise) [inline MainModule/Server/Commands/Players.lua:1153]: Does this actually do anything though

Comment (LiveFireExercise) [reply MainModule/Server/Commands/Players.lua:1153]: The only plave I have ever seen this used is in settings.Commands

Comment (LiveFireExercise) [reply MainModule/Server/Commands/Players.lua:1153]: Alright so I would merge this but you haven't updated the SettingsDescriptions along with changing the settings

Comment (moo1210) [reply MainModule/Server/Commands/Players.lua:1153]: yes
https://github.com/Epix-Incorporated/Adonis/blob/1acb8016c55ac2a8bf14983b5cf0baa41a52c09c/MainModule/Server/Core/Admin.lua#L1460-L1462

Comment (LiveFireExercise) [reply MainModule/Server/Commands/Players.lua:1153]: Huh. I guess it does then

Comment (LiveFireExercise) [reply MainModule/Server/Commands/Players.lua:1153]: image
So it works, but it doesn't show up on the ingame settings menu

**Comment by moo1210** [comment]
https://github.com/Epix-Incorporated/Adonis/pull/1221/commits/c75af6806035b1bb42ad66922cfd524b69d2f9b8
https://github.com/Epix-Incorporated/Adonis/pull/1221/commits/c61d2017763ee92b7f8f3ec74eff89f575c04518

Thank you :pray:, I was close to finish writing a response to say to do basically that. There's really no reason to hold up a security-based PR over a detail with the in-game settings that the maintainer can extremely easily just fix right then before merging, since they already mentioned it.

**Comment by LiveFireExercise** [comment]
image
image

Unfortunately it still doesn't seem to show up. Maybe I am an idiot

**Comment by moo1210** [comment]
you didn't add it to the order table

**Comment by LiveFireExercise** [comment]
i always forget about the order table. aka the most forgettable part of the settings module

**Comment by LiveFireExercise** [comment]
Aight, in a historic first after testing it everything seems to work

**Comment by alau740** [comment]
a first for adonis

**Comment by LiveFireExercise** [comment]
truly a momentus occasion

**Comment by P3tray** [comment]
> Aight, in a historic first after testing it everything seems to work

my adonis fails to even run !!!!11!

**Comment by LiveFireExercise** [comment]
You problem

**Pull request merged**